sematic/resolvers/silent_resolver.py

```diff
@@ -50,3 +63,85 @@ def _end_inline_execution(self, future_id) -> None:
 
     def _wait_for_scheduled_runs(self) -> None:
         pass
+
+    def _resolution_did_fail(self, error: Exception) -> None:
+        self._deactivate_all_resources()
+
+    def _deactivate_all_resources(self) -> None:
```

Comment:
quite possibly. For example, it might be used for cancellation in the local resolver (silent resolver doesn't support cancellation, IIRC). Why *not* have it be its own function? Seems more readable to be anyway since you don't have to look in `_resolution_did_fail` and read the loop to realize "oh, this is deactivating all the resources."